### ISSUE TITLE
Добавить сущность "объект"

### ISSUE THREAD

**Issue by sergio-fry** [HeavyTechRuby/history#55]
Сейчас адрес вводится в текстовую строку и каждая история хранится отдельно от других, даже если адрес один и тот же

Необходимо добавить сущность "объект", и при вводе адреса пытаться найти уже существующий. Если такого нет, то добавлять новый. На главную нужно вывести список последних добавленных объектов. Страница объекта должна содержать список последних историй по этому объекту.

**Comment by o-200** [comment]
> @o-200 не готова часть
> 
> > при вводе адреса пытаться найти уже существующий. Если такого нет, то добавлять новый

При создании истории если нету существующего адреса (локации), то создаётся новая локация с тем адресом. 

Разве это не оно?

```ruby
  before_validation :set_location

  def set_location
    location = Location.find_by(address: self.address)

    if location
      self.location = location
    else
      self.location = Location.create(address: self.address)
    end
  end
  ```

**Comment by sergio-fry** [comment]
Это оно, но не совсем. Тут имеется ввиду, что при есть поле с автоподстановкой по адресу. Можно создать новую issue под это. А то большая часть, действительно уже сделана. Могу тебя попросить сделать задачку?

**Comment by o-200** [comment]
> Это оно, но не совсем. Тут имеется ввиду, что при есть поле с автоподстановкой по адресу. Можно создать новую issue под это. А то большая часть, действительно уже сделана. Могу тебя попросить сделать задачку?

Да, без проблем. Но я занимаюсь над #57 и брать две задачи не хочу. Возможно, кто-нибудь другой захочет доделать задачу.

Предлагаю поступить так - если после того как я доделаю #57 не найдётся исполнителя на эту задачу, то уже я приступлю к завершению задачи.

**Comment by o-200** [comment]
решил доделать эту часть

**Comment by o-200** [comment]
@sergio-fry так как нужно делать выпадающий список, то из этого выявлена проблема - нужно прикрутить запуск javascript части приложения. 

Объём задач будет большим, считаю стоит создать новую задачу под это.

**Comment by realvint** [comment]
Возьми библиотеку tom-select. Весь нужный функционал можно реализовать через неё

**Comment by realvint** [comment]
> На данный момент используется предустановленный sprockets, но он не запускается
Куда он не запускается?))
В этом проекте НЕТ nodejs
Причём тут докер?))
Изучи как работает importmaps

**Comment by realvint** [comment]
> при запуске приложения не запускается javascript

...

**Comment by o-200** [comment]
> > На данный момент используется предустановленный sprockets, но он не запускается
> 
> Куда он не запускается?)) В этом проекте НЕТ nodejs Причём тут докер?)) Изучи как работает importmaps

Ok

**Comment by sergio-fry** [comment]
@o-200 на какие задачи предлагаешь разделить?

**Comment by o-200** [comment]
> @o-200 на какие задачи предлагаешь разделить?

Я думаю как минимум нужно сделать отдельную задачу на поиск объекта при его создании, так как в описании к задаче не было упоминания такого функционала.

Сложность тут в том чтобы понять как лучше делать таковой поиск на бэкенде, наверное стоит обсудить этот вопрос. Подключаем поисковый движок по типу elastic search или просто делаем через полнотекстовый поиск?

Я бы обсудил ещё смену importmap на какой-нибудь esbuild, но это вкусовщина и если обязательно использовать именно импортные карты - ок
